application/src/GameLogic.cpp

```diff
@@ -1,11 +1,41 @@
+#include <iostream>
 #include "GameLogic.h"
 
 GameLogic::GameLogic() {
     this->state = mainMenu;
+    this->walrus1 = std::unique_ptr<Player>(new Player());
+    this->walrus2 = std::unique_ptr<Player>(new Player());
 }
 
 void GameLogic::update(float dSec) {
 
+    this->walrus1->tickMovement(dSec);
+    this->walrus2->tickMovement(dSec);
+    //std::cout << this->walrus1->pos.x << "," << this->walrus1->pos.y << "  " <<  this->walrus1->vel.x << "," << this->walrus1->vel.y << std::endl;
+    //this->walrus2.pos += (this->walrus2.vel) * dSec;
+    if (this->walrus1->vel.x > 0) {
```

Comment:
this friction deceleration code should probably be it's own function (sort of what I had imagined with ApplyPassiveForce)